app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt

```diff
@@ -955,113 +996,700 @@ class ExplorationActivityTest {
+      )
+      testCoroutineDispatchers.runCurrent()
+
+      pressBack()
+
+      onView(withText(R.string.progress_database_full_dialog_back_to_lesson_button))
+        .inRoot(isDialog()).perform(click())
+
+      explorationCheckpointTestHelper.verifyExplorationProgressIsSaved(
+        internalProfileId,
+        FRACTIONS_EXPLORATION_ID_0
+      )
+      explorationCheckpointTestHelper.verifyExplorationProgressIsSaved(
+        internalProfileId,
+        FAKE_EXPLORATION_ID_1
+      )
+      explorationCheckpointTestHelper.verifyExplorationProgressIsSaved(
+        internalProfileId,
+        FAKE_EXPLORATION_ID_2
+      )
+    }
+    explorationDataController.stopPlayingExploration()
+  }
+
+  private fun createExplorationActivityIntent(
+    internalProfileId: Int,
+    topicId: String,
+    storyId: String,
+    explorationId: String,
+    shouldSavePartialProgress: Boolean
+  ): Intent {
+    return ExplorationActivity.createExplorationActivityIntent(
+      ApplicationProvider.getApplicationContext(),
+      internalProfileId,
+      topicId,
+      storyId,
+      explorationId,
+      /* backflowScreen= */ null,
+      shouldSavePartialProgress
+    )
+  }
+
+  private fun setupAudio() {
+    // Only initialize the Robolectric shadows when running on Robolectric (and use reflection since
+    // Espresso can't load Robolectric into its classpath).
+    if (isOnRobolectric()) {
+      val dataSource = createAudioDataSource(
+        explorationId = RATIOS_EXPLORATION_ID_0, audioFileName = "content-en-057j51i2es.mp3"
+      )
+      addShadowMediaPlayerException(dataSource!!, IOException("Test does not have networking"))
+    }
+  }
+
+  private fun setUpAudioForFractionLesson() {
+    // Only initialize the Robolectric shadows when running on Robolectric (and use reflection since
+    // Espresso can't load Robolectric into its classpath).
+    if (isOnRobolectric()) {
+      val dataSource = createAudioDataSource(
+        explorationId = FRACTIONS_EXPLORATION_ID_0, audioFileName = "content-en-nb3k4zuyir.mp3"
+      )
+      val dataSource2 = createAudioDataSource(
+        explorationId = FRACTIONS_EXPLORATION_ID_0, audioFileName = "content-hi-en-l8ik9pdxj2a.mp3"
+      )
+      addShadowMediaPlayerException(dataSource!!, IOException("Test does not have networking"))
+      addShadowMediaPlayerException(dataSource2!!, IOException("Test does not have networking"))
+    }
+  }
+
+  private fun addShadowMediaPlayerException(dataSource: Any, exception: Exception) {
+    val classLoader = ExplorationActivityTest::class.java.classLoader!!
+    val shadowMediaPlayerClass = classLoader.loadClass("org.robolectric.shadows.ShadowMediaPlayer")
+    val addException =
+      shadowMediaPlayerClass.getDeclaredMethod(
+        "addException", dataSource.javaClass, IOException::class.java
+      )
+    addException.invoke(/* obj= */ null, dataSource, exception)
+  }
+
+  private fun scrollToViewType(viewType: StateItemViewModel.ViewType) {
```

Comment:
remove this function as this is not getting used by any function

Reply:
done